feat: Add libadwaita

danyeaw opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Zulip chat][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://conda-forge.zulipchat.com

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/libadwaita/meta.yaml```) and found it was in an excellent condition.

danyeaw commented:
Depends on https://github.com/conda-forge/gtk4-feedstock/pull/66.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint --conda-forge .` from the recipe directory. You can also examine the [workflow logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/20629870960) for more detail.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/20629870960](https://github.com/conda-forge/conda-forge-webservices/actions/runs/20629870960). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/libadwaita/recipe.yaml```, ```recipes/appstream/recipe.yaml```, ```recipes/libxmlb/recipe.yaml```, ```recipes/libsass-native/recipe.yaml```, ```recipes/sassc/recipe.yaml```, ```recipes/itstool/recipe.yaml```) and found it was in an excellent condition.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/libsass-native/recipe.yaml```, ```recipes/appstream/recipe.yaml```, ```recipes/libadwaita/recipe.yaml```, ```recipes/sassc/recipe.yaml```, ```recipes/libxmlb/recipe.yaml```, ```recipes/itstool/recipe.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/libxmlb/recipe.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-recipe-manager`. The recipe can only be automatically migrated to the new v1 format if it is parseable by conda-recipe-manager.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/20650545695](https://github.com/conda-forge/conda-forge-webservices/actions/runs/20650545695). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/sassc/recipe.yaml```, ```recipes/libadwaita/recipe.yaml```, ```recipes/libsass-native/recipe.yaml```, ```recipes/itstool/recipe.yaml```, ```recipes/libxmlb/recipe.yaml```, ```recipes/appstream/recipe.yaml```) and found it was in an excellent condition.

danyeaw commented:
Hey @haecker-felix, thanks so much for the additional testing and the feedback. I added appstream and libxmlb to the run_exports, let me know if that looks good to you. Also thanks for volunteering to help maintain libadwaita as well, I added you as a maintainer.

pkgw commented:
@danyeaw I haven't reviewed everything in depth, but this PR is looking sensible to me at a high level. I'd be OK with merging it.

haecker-felix commented:
@danyeaw thanks for adding me as maintainer 👍. For your information, there's a MR that's worth keeping an eye on: https://gitlab.gnome.org/GNOME/libadwaita/-/merge_requests/1676

Maybe it would allow as in future to no longer depend on `appstream`, which would mean that we could build libadwaita on Windows without having to patch it. 

And regarding the run exports, I'm not entirely sure what's the best practice for that, but I think I would add 

```
  run_exports:
    - ${{ pin_subpackage('libadwaita', upper_bound='x.x') }}
```

to libadwaita

```
  run_exports:
    - ${{ pin_subpackage('libxmlb', upper_bound='x.x') }}
```

to libxmlb, and 

```
  run_exports:
    - ${{ pin_subpackage('appstream', upper_bound='x.x') }}
```

to appstream, then everything should get pulled in correctly automatically. For example, right now you added libxmlb to libadwaita, even though it being an appstream dependency.
